### PR TITLE
Python 2/3: handle lambda (x,y) + update notebooks

### PR DESCRIPTION
**Pull request by gpotter2**
This PR:
- updates the notebooks with the new lambda format `lambda x,y: ...` instead of `lambda (x,y): ...`
- updates ploting functions to support both formats on python 2 (on python 3, the first one isn't available)

### CODE REVIEW

**Comment by karpierz** [comment]
Take a look at #1288

File: scapy/compat.py [hunk 1]
```diff
@@ -48,6 +48,21 @@ def cmp(a, b):
...
 
+if six.PY2:
+    def lambda_tuple_converter(func):
+        """Converts a Python 2 function as
+            lambda (x,y): x + y
+        In the Python 3 format:
+            lambda x,y : x + y"""
+        if func is None:
+            return func
+        if func.__code__.co_varnames[0] == ".0": # Detected format lambda (x,y): ...
```
[the remark this answers is not in the document]
Comment (gpotter2) [reply scapy/compat.py]: Tests are passing. Could you test ?

File: scapy/compat.py [hunk 2]
```diff
@@ -48,6 +48,18 @@ def cmp(a, b):
...
+    Converts a Python 2 function as
+      lambda (x,y): x + y
+    In the Python 3 format:
+      lambda x,y : x + y
+    """
+    if func is None:
+        return func
+    if func.__code__.co_argcount == 1: # Detected format lambda (x,y): ... or lambda x: x[0]...
+        return lambda *args: func(args[0] if len(args) == 1 else args)
+    return func
```
Comment (karpierz) [inline scapy/compat.py]: if func is not None and  func.__code__.co_argcount == 1:
    return lambda *args: func(args[0] if len(args) == 1 else args)
else:
    return func

**Comment by karpierz** [comment]
Yes, now is much better :).

File: test/regression.uts [hunk 3]
```diff
@@ -1100,7 +1100,13 @@ old_debug_dissector = conf.debug_dissector
...
 ans,unans=sr(IP(dst="www.google.com/30")/TCP(dport=[80,443]),timeout=2)
 conf.debug_dissector = old_debug_dissector
-ans.make_table(lambda s_r: (s_r[0].dst, s_r[0].dport, s_r[1].sprintf("{TCP:%TCP.flags%}{ICMP:%ICMP.code%}")))
+
+# Backward compatibility: Python 2 only
+if six.PY2:
+    exec("""ans.make_table(lambda (s, r): (s.dst, s.dport, r.sprintf("{TCP:%TCP.flags%}{ICMP:%ICMP.code%}")))""")
+
+# New format: all Python versions
+ans.make_table(lambda s, r: (s.dst, s.dport, r.sprintf("{TCP:%TCP.flags%}{ICMP:%ICMP.code%}")))
```
[the remark this answers is not in the document]
Comment (karpierz) [reply test/regression.uts]: Why it did not add?

**Comment by gpotter2** [comment]
Can't access my git repos / emails for now (server crashed)...

**Comment by gpotter2** [comment]
@guedou Rebased & fixed flake8. Any news on this PR?

**Comment by guedou** [comment]
Waiting for @p-l- approval.

**Comment by gpotter2** [comment]
Rebased & fixed conflicts (again)

**Comment by guedou** [comment]
There is another conflict =/

**Comment by guedou** [comment]
@karpierz I don't understand your comment.

**Comment by karpierz** [comment]
> -- lambdas must be written using a single argument when using tuples: use `lambda x_y: x_y[0] + f(x_y[1])` instead of `lambda (x, y): x + f(y)`.  
> +- lambdas must be written using a single argument when using tuples: use `lambda x, y: x + f(y)` instead of `lambda (x, y): x + f(y)`.  

This exclamation is currently wrong/outdated.

**Pull request merged**